front/lib/api/assistant/generation.ts

```diff
@@ -185,6 +185,12 @@ export async function renderConversationForModel({
     if (tokensUsed + c <= allowedTokenCount) {
       tokensUsed += c;
       selected.unshift(messages[i]);
+    } else if (messages[i].role === "content_fragment") {
+      const c = allowedTokenCount - tokensUsed;
+      tokensUsed += c;
+      const truncatedMessage = messages[i];
```

Comment:
Why instantiate this variable? this won't clone it right? so mutating its content will mutate the original no?

Reply:
This is fine since we have created these objects but the code can be much shorter no?

Reply:
This was just for clarity but if it add confusion, I will just remove it.